packages/resolve-dependencies/src/index.ts

```diff
@@ -173,7 +174,15 @@ export default async function (
       }
     }
   }
-  const { newLockfile, pendingRequiresBuilds } = updateLockfile(dependenciesGraph, opts.wantedLockfile, opts.virtualStoreDir, opts.registries) // eslint-disable-line:prefer-const
+
+  // If --fix-lockfile is specified, the previous lockfile's packages is ignored
+  // and the packages field(all field execept version) will be generated from the new resolved depNode
+  const { newLockfile, pendingRequiresBuilds } = updateLockfile(dependenciesGraph, opts.fixLockfile
+    ? {
+      importers: opts.wantedLockfile.importers,
+      lockfileVersion: opts.wantedLockfile.lockfileVersion,
+    }
+    : opts.wantedLockfile, opts.virtualStoreDir, opts.registries) // eslint-disable-line:prefer-const
```

Comment:
But initially your issue was that you could not fix the existing lockfile. This feature is not very useful if you just ignore the existing lockfile. You could just remove the lockfile and run `pnpm install`.

Reply:
Remove the lockfile and run pnpm install do not preserve the existing locked version.  

Here we can preserve the locked  version, and regenerate the missing field,    other fields will mostly remains the same if locked version does not change